src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -111,12 +111,24 @@ export function makeUtxoEngineState(
 
   const clearTaskCache = (): void => {
     taskCache.blockWatching = false
-    taskCache.addressSubscribeCache = {}
-    taskCache.addressTransactionCache = {}
-    taskCache.addressUtxoCache = {}
-    taskCache.rawUtxoCache = {}
-    taskCache.processorUtxoCache = {}
-    taskCache.updateTransactionCache = {}
+    for (const key of Object.keys(taskCache.addressSubscribeCache)) {
+      removeItem(taskCache.addressSubscribeCache, key)
+    }
+    for (const key of Object.keys(taskCache.addressTransactionCache)) {
+      removeItem(taskCache.addressTransactionCache, key)
+    }
+    for (const key of Object.keys(taskCache.addressUtxoCache)) {
+      removeItem(taskCache.addressUtxoCache, key)
+    }
+    for (const key of Object.keys(taskCache.rawUtxoCache)) {
+      removeItem(taskCache.rawUtxoCache, key)
+    }
+    for (const key of Object.keys(taskCache.processorUtxoCache)) {
+      removeItem(taskCache.processorUtxoCache, key)
+    }
+    for (const key of Object.keys(taskCache.updateTransactionCache)) {
+      removeItem(taskCache.updateTransactionCache, key)
+    }
```

Comment:
Since you are removing everything anyway, there is no benefit in doing this other then maybe type safety but it's not really needed here anyway.
Revert this commit please.

Reply:
It's needed to eliminate object reference bugs.